text/0000-temp-config-settings.md

```diff
@@ -13,65 +13,376 @@ nit-picking the "gotchas" in code examples. :heart:
+
+```js
+include: ['./src/**/*.{js,jsx,ts,tsx}']
+```
+
+### exclude - `[]`
+
+Same as include.
+
+### conditions - `Extendable<PandoConditions>`
+
+Here we will define all the conditions for Pando usage. An example would be:
+
+```js
+conditions: {
+  light: `[data-color-mode="light"] &`,
+  dark: `[data-color-mode="dark"] &`,
+  system: `[data-color-mode="system"] &`,
+  pando: `[data-theme="pando"] &`,
+
+  extend: {
+    showNav: '&:is([data-show-nav="true"])'
+  },
+},
+```
+
+This allows our theming to be even more flexible for a feature set by allowing the
+user to choose a `mode` and a `theme` option. For example, in a global nav/app-bar,
+the user can always toggle "light" or "dark" mode, but in their profile settings
+they can also choose any theme (i.e. "pando", "pizza-site", "classic", etc.).
+
+### globalCSS - `Extendable<PandoGlobalStyles>`
+
+This is where all remaining normalize styles would go that is not handled by
+recipes or patterns.
+
+### theme - `Extendable<PandoTheme>`
+
+This is where all the tokens would go. This would be the same as the current
+`design-tokens` package except use a object structure that has a better developer experience.
+
+```js
+bg="default.surface.weak"
+bg="action.bg"
+bg="danger.bg"
+borderColor="info.border"
+borderColor="warning.border"
+borderColor="success.border"
+```
+
+Another example using the Panda `css` pattern:
+
+```jsx
+<Button className={css({
+  _hover: {
+    bg: 'action.bg-hover',
+  },
+})} />
+```
+
+This would also define our fonts (both PS and code).
+
+### utilities - `Extendable<PandoUtilities>`
+
+This is where all the utilities would go. Here is an example of a good one that
+creates a shorthand prop for `padding-inline` we would include:
+
+```js
+pse: {
+  className: 'padding-inline-start',
+  values: 'spacing',
+  transform(value: CssValue<string>) {
+    return {
+      paddingInlineStart: value,
+      paddingInlineEnd: value,
+    }
+  },
+},
+```
+
+With the above, you can now streamline your code by using the following:
+
+```jsx
+<div className={css({ pse: '4' })} />
+```
+
+### patterns - `Extendable<PandoPatterns>`
+
+This is where all the patterns would go if we need them. I can't think of any
+off the top of my head, but I'm sure we will need some?
+
+### jsxFramework = `react`
+
+This will allow Panda to know how to build the jsx components for teams that need
+a custom solution outside of our React library.
+
+> :warning: **Note:** These components will **not** be accessible. It is therefore
+> recommended that teams use our React library for the "end all solution".
+
+### jsxFactory = `pando`
+
+This will create the jsx components to use the naming convention of:
+
+```jsx
+<pando.div />
+```
 
 # Drawbacks
 
-Why should we *not* do this? Please consider:
+With any change, there are tradeoffs. Especially when trying to create structure
+where there is none.
+
+## Why should we _not_ do this?
```

Comment:
I'm definitely a proponent of leveraging third-party OS libraries that are stable and widely supported, but in the past you've raised concerns when that model was suggested for delivering Pando features. e.g. using Radix or Chakra to provide the accessible React primitives styled by Pando.

I'm paraphrasing because I can't find that discussion, but I think the argument was generally to keep our solution independent, nimble and tailored to our needs and standards. Do those concerns apply to Panda or is it a different paradigm?

Reply:
@timkinnane great question - it's different in paradigm due to the nature that it's a library essentially *for* proving foundations for either library's or apps. Essentially, it's literally what and where we were going for HS, so it just doesn't make sense for us to waste time on that anymore.

In the contrast - using something like Chakra is too much of that foundation and creates too much ownership over decisions we need the IP over. This is specifically related to comms and our business IP to massively promote Pando on all social media outlets when we get to v1. Panda is that "sweet spot" you could say that doesn't cross the line while also giving us "just enough" of a foundation without forcing us to conform to an overall design.

Reply:
The bigger reason though (IMO) over the business needs is that it is a really great framework (and you know I don't like Chakra) and I have yet to find a use case that either:

1. Hasn't improve my code-base
2. Is not possible to do
3. Is not surprisingly joyful to use

So, needing something that brings teams who are all over the place regarding libraries together - it's the perfect solution since it's purpose is basically the marriage of CSS-in-JS, Chakra, and Tailwind.